index.js

```diff
@@ -379,6 +383,17 @@ module.exports = class NoiseSecretStream extends Duplex {
     const id = buf.subarray(3, 3 + 32)
     streamId(handshakeHash, this.isInitiator, id)
 
+    // setup secretbox state for unordered messages
+    this._boxSecret = b4a.allocUnsafeSlow(32)
+    this._boxNonceEncrypt = b4a.allocUnsafeSlow(sodium.crypto_secretbox_NONCEBYTES)
+    this._boxNonceDecrypt = b4a.allocUnsafeSlow(sodium.crypto_secretbox_NONCEBYTES)
+
+    sodium.crypto_generichash(this._boxSecret, NS_BOX, handshakeHash)
+    sodium.crypto_generichash(this._boxNonceEncrypt, b4a.concat([NS_BOX, publicKey]), handshakeHash)
+    sodium.crypto_generichash(this._boxNonceDecrypt, b4a.concat([NS_BOX, remotePublicKey]), handshakeHash)
+    b4a.fill(this._boxNonceDecrypt, 0, 0, 8) // zerofill first 8 bytes
+    b4a.fill(this._boxNonceEncrypt, 0, 0, 8)
```

Comment:
instead of resetting this to 0, should be just save the initial buffer and use that for the overflow check instead? less 0000 bits over the wire

Reply:
cc @chm-diederichs on this also